dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.rpm.Rpm.xml

```diff
@@ -212,6 +212,23 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
         <arg name="options" type="a{sv}" direction="in" />
     </method>
 
+    <!--
+        system_upgrade:
+        @options: an array of key/value pairs to modify system_upgrade behavior
+
+        Prepare a transaction for upgrade to the new distribution release. The prepared transaction should be executed during reboot (see `offline` option of the `do_transaction` method of the `Goal` interface).
+
+        Following @options are supported:
+
+            - mode: string (one of "distrosync", "upgrade", default is "distrosync")
+                By default the system_upgrade behaves like `dnf distro-sync`, always installing packages from the new release, even if they are older then the currently installed version. If set to "upgrade", packages from the new release are not installed if they are older then what is currently installed (behave like `dnf upgrade`).
```

Comment:
Are the similarities with those modes the `dnf5 distro-sync` and `dnf5 system-upgrade`? I see some differences between the two, like the first can run "online" or "offline" while the other is always "offline". The first has `--download only`, the second is always split into two phases, `download` and `reboot`.

For the gnome-software it'll be always `download` and then `reboot-to-install`. It should be possible to change my mind and not install previously prepared system upgrade.

Reply:
It's more about what method `system-upgrade` uses in background. By default the `dnf5 system-upgrade download` is roughly equivalent to `dnf5 distro-sync --offline`. But with `dnf5 system-upgrade download --no-downgrade` option the system upgrade makes sure, that packages from the new release that are of older version then currently installed are not used. In this case its roughly equivalent to `dnf5 upgrade --offline`.

I will add code to enforce "offline" behavior to "Rpm.system_upgrade()" call, because currently it relies on `offline` option being passed to `Goal.do_transaction()` method.